forge-gui/res/cardsfolder/upcoming/snarlfang_vermin.txt

```diff
@@ -0,0 +1,11 @@
+Name:Snarlfang Vermin
+ManaCost:B
+Types:Creature Rat
+PT:2/1
+T:Mode$ DamageDone | CombatDamage$ True | ValidSource$ Card.Self | ValidTarget$ Creature | TriggerZones$ Battlefield | Execute$ TrigSuspect | TriggerDescription$ Whenever CARDNAME deals combat damage to a creature, if that creature is still on the battlefield, suspect that creature. It perpetually gains this ability.
+SVar:TrigSuspect:DB$ AlterAttribute | Defined$ TriggeredTargetLKICopy | ConditionDefined$ TriggeredTargetLKICopy | ConditionPresent$ Card.inZoneBattlefield | RememberObjects$ TriggeredTargetLKICopy | Attributes$ Suspected | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ TriggeredTargetLKICopy | ConditionDefined$ TriggeredTargetLKICopy | ConditionPresent$ Card.inZoneBattlefield | Triggers$ CombatTrig | Duration$ Perpetual
+SVar:CombatTrig:Mode$ DamageDone | CombatDamage$ True | ValidSource$ Card.Self | ValidTarget$ Creature | TriggerZones$ Battlefield | Execute$ TrigSuspect | TriggerDescription$ Whenever CARDNAME deals combat damage to a creature, if that creature is still on the battlefield, suspect that creature. It perpetually gains this ability.
+T:Mode$ ChangesZone | Origin$ Battlefield | PresentZone$ Graveyard | IsPresent$ Card.StrictlySelf | TriggerZones$ Graveyard | Destination$ Graveyard | ValidCard$ Creature.OppCtrl+IsSuspected | Execute$ TrigLoseLife | TriggerDescription$ Whenever a suspected creature an opponent controls dies while CARDNAME is in your graveyard, that opponent loses 1 life.
```

Comment:
you scripted intervening if here

Reply:
Do you mean that` IsPresent$ Card.StrictlySelf |` isn't necesairy here ?

Reply:
You applied this rule:
> 603.4. A triggered ability may read "When/Whenever/At [trigger event], if [condition], [effect]." When the trigger event occurs, the ability checks whether the stated condition is true. The ability triggers only if it is; otherwise it does nothing. If the ability triggers, it checks the stated condition again as it resolves. If the condition isn't true at that time, the ability is removed from the stack and does nothing. Note that this mirrors the check for legal targets. This rule is referred to as the "intervening 'if' clause" rule.

but nothing in your trigger matches this templating, so it's only a normal trigger event instead

Reply:
I'm not sure i'm exactly following how this is scripted like an intervening if. I've used Boneyard Scourge as a template which is worded like this card

Reply:
yea that one needs to be fixed by me
IsPresent will get checked on resolve so you need NoResolvingCheck

Reply:
I think I'm only confusing you more at this point since you removed the condition now again :P
but it shouldn't matter too much for Alchemy and my fix there